src/EXTRA-DUMP/dump_extxyz.cpp

```diff
@@ -0,0 +1,259 @@
+    }
+  }
+}
+
+/* ---------------------------------------------------------------------- */
+
+void DumpExtXYZ::init_style()
+{
+  if (!typenames)
+    error->all(FLERR, Error::NOLASTLINE,
+               "Must use either typelables or dump_modify element with dump style extxyz");
+
+  DumpXYZ::init_style();
+  update_properties();
+}
+
+/* ---------------------------------------------------------------------- */
+
+int DumpExtXYZ::modify_param(int narg, char **arg)
+{
+  int rv = DumpXYZ::modify_param(narg, arg);
+  if (rv > 0) return rv;
+
+  if (strcmp(arg[0],"vel") == 0) {
+    if (narg < 2) error->all(FLERR,"Illegal dump_modify command");
+    with_vel = utils::logical(FLERR,arg[1],false,lmp);
+    update_properties();
+    return 2;
+  }
+
+  if (strcmp(arg[0],"forces") == 0) {
+    if (narg < 2) error->all(FLERR,"Illegal dump_modify command");
+    with_forces = utils::logical(FLERR,arg[1],false,lmp);
+    update_properties();
+    return 2;
+  }
+
+  if (strcmp(arg[0],"mass") == 0) {
+    if (narg < 2) error->all(FLERR,"Illegal dump_modify command");
+    with_mass = utils::logical(FLERR,arg[1],false,lmp);
+    update_properties();
+    return 2;
+  }
+
+  return 0;
+}
+
+/* ---------------------------------------------------------------------- */
+
+void DumpExtXYZ::write_header(bigint n)
+{
+  if (me == 0) {
+    if (!fp) error->one(FLERR, Error::NOLASTLINE, "Must not use 'run pre no' after creating a new dump");
+
+    std::string header;
+
+    header = fmt::format("{}\nTimestep={}", n, update->ntimestep);
+    if (time_flag) header += fmt::format(" Time={:.6f}", compute_time());
+    header += fmt::format(" pbc=\"{} {} {}\"", domain->xperiodic ? "T" : "F",
+                          domain->yperiodic ? "T" : "F", domain->zperiodic ? "T" : "F");
+    header +=
+        fmt::format(" Lattice=\"{:g} {:g} {:g} {:g} {:g} {:g} {:g} {:g} {:g}\"", domain->xprd, 0.,
+                    0., domain->xy, domain->yprd, 0., domain->xz, domain->yz, domain->zprd);
+    header += fmt::format(" Properties={}", properties_string);
+    utils::print(fp, header + "\n");
+  }
+}
+
+/* ---------------------------------------------------------------------- */
+
+void DumpExtXYZ::pack(tagint *ids)
+{
+  int m, n;
+
+  tagint *tag = atom->tag;
+  int *type = atom->type;
+  int *mask = atom->mask;
+  double **x = atom->x;
+  double **v = atom->v;
+  double **f = atom->f;
+  double *mass = atom->mass;
+  int nlocal = atom->nlocal;
+
+  m = n = 0;
+  for (int i = 0; i < nlocal; i++)
+    if (mask[i] & groupbit) {
+      buf[m++] = tag[i];
+      buf[m++] = type[i];
+      buf[m++] = x[i][0];
+      buf[m++] = x[i][1];
+      buf[m++] = x[i][2];
+      if (with_vel) {
+        buf[m++] = v[i][0];
+        buf[m++] = v[i][1];
+        buf[m++] = v[i][2];
+      }
+      if (with_forces) {
+        buf[m++] = f[i][0];
+        buf[m++] = f[i][1];
+        buf[m++] = f[i][2];
+      }
+      if (with_mass) {
+        buf[m++] = mass[type[i]];
```

Comment:
```suggestion
        if (rmass) {
          buf[m++] = rmass[i];
        } else {
          buf[m++] = mass[type[i]];
        }
```
This is for compatibility with atom styles that have per-atom masses.